### PR TITLE
feat(app): update to next 9 and styled-components 4

### PR DESCRIPTION
**Pull request by douglasduteil**
---

Following #83

### CODE REVIEW

**Comment by douglasduteil** [comment]
Works on IE11 :kiss:

File: packages/app/package.json [hunk 1]
```diff
@@ -59,16 +60,18 @@
...
-    "eslint": "^5.16.0",
+    "eslint": "^6.1.0",
     "eslint-config-prettier": "^4.2.0",
     "eslint-plugin-prettier": "^3.0.1",
     "eslint-plugin-react": "^7.14.2",
-    "gh-pages": "^2.0.1",
     "jest-cli": "^24.8.0",
     "json-loader": "^0.5.7",
+    "lodash.flow": "^3.5.0",
+    "next-transpile-modules": "^2.3.1",
```
Comment (revolunet) [inline packages/app/package.json]: Not needed?

File: packages/app/package.json [hunk 2]
```diff
@@ -59,16 +60,18 @@
...
     "postcss-loader": "^3.0.0",
     "prettier": "^1.17.0",
     "redux-devtools-extension": "^2.13.8",
-    "style-loader": "^0.23.1"
+    "style-loader": "^0.23.1",
+    "webpack": "^4.37.0"
   }
```
Comment (revolunet) [inline packages/app/package.json]: Isnt webpack included with nexr?

File: packages/app/pages/_document.js [hunk 3]
```diff
@@ -1,48 +1,56 @@
...
+      return {
+        ...initialProps,
+        styles: (
+          <>
+            {initialProps.styles}
+            {sheet.getStyleElement()}
+          </>
+        )
+      };
+    } finally {
```
Comment (revolunet) [inline packages/app/pages/_document.js]: Maybe add a catch to capture exceptions ?

**Comment by revolunet** [comment]
I think we should better replace the 3 faulty modules with some browser-friendly ones rather than add non-standard scripts

File: packages/app/package.json (hunk 2 above)
Comment (douglasduteil) [reply packages/app/package.json]: It's a peerdependency

**Comment by revolunet** [comment]
candidates :

 - query-string : https://github.com/SpainTrain/querystring-es3
 - split-on-first : looks eS3 compatible ? https://github.com/sindresorhus/split-on-first/blob/master/index.js
 - strict-uri : https://www.npmjs.com/package/uri-js

File: packages/app/package.json (hunk 2 above)
Comment (douglasduteil) [reply packages/app/package.json]: Adding it removes lots of warnings

```
warning "workspace-aggregator-2969c09f-023d-453e-ae0c-aaddf4877b4b > @emjpm/app > @zeit/next-css > css-loader@1.0.0" has unmet peer dependency "webpack@^4.0.0".
warning "workspace-aggregator-2969c09f-023d-453e-ae0c-aaddf4877b4b > @emjpm/app > @zeit/next-css > mini-css-extract-plugin@0.4.3" has unmet peer dependency "webpack@^4.4.0".
warning "workspace-aggregator-2969c09f-023d-453e-ae0c-aaddf4877b4b > @emjpm/app > next-images > file-loader@4.0.0" has unmet peer dependency "webpack@^4.0.0".
warning "workspace-aggregator-2969c09f-023d-453e-ae0c-aaddf4877b4b > @emjpm/app > next-images > url-loader@2.0.1" has unmet peer dependency "webpack@^4.0.0".
```

**Pull request merged**